src/inverted-index.js

```diff
@@ -1,160 +1,143 @@
-        var results = [];
-        // Checking if arguments have been passed
-        if (arguments.length) {
-            //  Format
-            var format = function (word) {
-                return word.toLowerCase()
-                .replace(/[,.?!]/g, '')
-                .split(' ')
-                .filter(function (element) {
-                    return stop_words.indexOf(element) === -1;
-                });
-            };
-            // Returning Index of search terms
-            var search = function (oneWord) {
-                found = false;
-                if (theIndex && oneWord) {
-                    theIndex.forEach(function (indexObject) {
-                        if (indexObject.hasOwnProperty(oneWord)) {
-                            found = true;
-                            results = results.concat(indexObject[oneWord]);
-                        }
-                    });
-
-                    if (!found) {
-                        results.push(-1);
-                    }
-                }
-            };
-
-            // 1. Loop through the args to get each term
-            // 2. Check the type of the term
-            // 3. If a string
-            // 4. Split the term in ' '
-            // 5. Loop through the splitted terms
-            // 6. Search for the term
-            // 7. Else if array:
-            // 8. loop through the array, toLowerCase() search
+  this.getIndex = function () {
+    return index;
+  }
 
-            var args = Array.from(arguments);
-            args.forEach(function (terms) {
-                if (typeof terms === 'string') {
-                    //  Format word to remove stop word and rswitch to lowercase
-                    var newTerm = format(terms);
-
-                    for (var word in newTerm) {
-                        search(newTerm[word]);
-                    }
-                } else if (Array.isArray(terms)) {
-                    // Loop through the array
-                    terms.forEach(function (element) {
-                        //  Format word to remove stop word and switch to lowercase
-                        element = format (element);
-                        // Search word
-                        element.forEach(function (oneWord) {
-                            search(oneWord);
-                        });
-                    });
-                } else {
-                    // Invalid search term
-                    results = false;
-                }
+  this.searchIndex = function () {
+    var theIndex = this.getIndex();
+    var results = {};
+    // Checking if arguments are passed
+    if(arguments.length) {
+      // Retutn Index of search terms
+      var search = function (oneWord) {
+        var found = false;
+        if (theIndex && oneWord) {
+          theIndex.forEach(function (indexObject) {
+            if(indexObject.hasOwnProperty(oneWord)) {
+              results[oneWord] = indexObject[oneWord];
```

Comment:
**Code Quality - Nitpick**

Results of search changed from an array to an object.